web_src/js/index.js

```diff
@@ -582,6 +582,25 @@ function initInstall() {
   });
 }
 
+function initIssueComments() {
+  if ($('.repository.view.issue .comments').length === 0) return;
+
+  $(document).click((event) => {
+    const $target = $(event.target);
+
+    if ($target.closest('.comment').length === 0) {
+      if (window.location.hash.length > 0) {
+        const i = window.location.toString().indexOf('#');
+
+        if (i >= 0) {
+          window.history.pushState({}, '', window.location.toString().substr(0, i));
+          window.location.reload();
```

Comment:
Instead of reload could you just modify CSS class?

Reply:
There is no css class. Its by :target selector

https://github.com/go-gitea/gitea/pull/9047/files#diff-12e7da6d8d2bc69c6b71119d1c9e6a96R844-R846

Reply:
Ha ha, I guess `<a href="#...."></a>` is out of fashion. 😄

Reply:
@techkowlogick and it does not refresh page if not highlighted.

Still i should change this behavior?

Reply:
Yes, please get rid of the reload. If you want to scroll to the anchor, use `element.scrollIntoView()`

Reply:
@silverwind what can i use instead of `:target` selector (to add/romve css class)?

Reply:
@silverwind I dont want to scroll. Why did you think that?

Reply:
I wonder what exactly this JS is trying to achieve as it seems to work just fine without any JS. Clicking the timestamp focuses the comment. Changing the active comment via both click or `location.hash = '#issue-id'` works.

What is that JS exactly good for?

Reply:
@silverwind it is for remove target path (#issuecomment-xx) from url on click outside of comment

Reply:
I'm concerned about the reload for accessibility purposes, as it will change location of screenreader. Additionally if someone has content in text box, the user may lose it (they'll be prompted by an alert box, but that may be confusing to user). This could also add load to the server by adding many additional requests.

Reply:
so we must change some behaviors? like

- on page load -> check url and add css class to comment
- on click outside of comment remove css

but i am not sure if this is correct way?

What is your suggestions

Reply:
What about `location.hash`? For example:

```
window.location.hash = "";  // clear hash
window.location.hash = "comment-33333";  // navigate to comment
```

Reply:
Check out how GitHub does it. It looks like they just replace the state of the history, which should remove the `:target`

Reply:
Something like this:

````js
const listener = (e) => {
  if (!window.location.hash) return;
  const target == document.querySelector(':target');
  if (!target || e.target.contains(target)) return;
  const scrollY = document.documentElement.scrollTop;
  window.history.replaceState(null, null, ' ');
  document.documentElement.scrollTop = scrollY;
  document.documentElement.removeEventListener('click', listener);
}
document.documentElement.addEventListener('click', listener);
```
This preserves scroll location on history reset, does not push useless history entries and does not require jQuery :)

Reply:
@guillep2k i tried it first then changed because it reset scroll position to 0. But works well if we set scrollposition again?

@jolheiser history.pushState(.. change url but does not affect anything :/

Reply:
@silver ooh i saw late your comment. What i did lastly is acceptable for your? or i can change as you suggested

Reply:
Would prefer if you can try my version. It may need a bit more refining, but doing it without jQuery would be quite nice.

Reply:
@silverwind

- replaceState does not remove :target affect
- we must handle click __only outside of 'targetted comment'__ 

I fnished. It is working. If you suggest raw js, i can change code